wgpu-core/src/command/compute.rs

```diff
@@ -476,8 +476,18 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             Some(&*query_set_guard),
         );
 
+        let label = if self
+            .instance
+            .flags
+            .contains(wgt::InstanceFlags::DISCARD_HAL_LABELS)
+        {
+            None
+        } else {
+            base.label
+        };
```

Comment:
Could we use `bool::then_some` here, and elsewhere?

Reply:
The closest thing I could do to what you are looking for is:

```rust
let label = (!self
    .instance
    .flags
    .contains(wgt::InstanceFlags::DISCARD_HAL_LABELS))
.then_some(base.label)
.flatten();
```

Which I find confusing in more ways than one (cargo fmt does a number on it). I'll add a helper function instead.

Reply:
Oh, sure, that's not an improvement. How about:

```
let label = base.label.filter(|_| !self.instance.flags.contains(wgt::InstanceFlags::DISCARD_HAL_LABELS));
```

JimB's Law: `Option` always has the right method.

But this is a very minor point, not worth delaying the PR for.